bugwarrior/services/jira.py

```diff
@@ -1,19 +1,49 @@
 from __future__ import absolute_import
 
-from jinja2 import Template
-from jira.client import JIRA
+from datetime import datetime
+
 import six
+from jinja2 import Template
+from jira.client import JIRA as BaseJIRA
+from requests.cookies import RequestsCookieJar
 
 from bugwarrior.config import asbool, die
 from bugwarrior.services import IssueService, Issue
 
+class ObliviousCookieJar(RequestsCookieJar):
+    def set_cookie(self, *args, **kwargs):
+        """Simply ignore any request to set a cookie."""
+        pass
+
+    def copy(self):
+        """Make sure to return an instance of the correct class on copying."""
+        return ObliviousCookieJar()
+
+
+# 'Borrowed' from this wonderful commit by @GaretJax
+# https://github.com/GaretJax/lancet/commit/f175cb2ec9a2135fb78188cf0b9f621b51d88977
+# Prevents Jira web client being logged out when API call is made.
```

Comment:
Can you add one more inline comment here that mentions the snippet is licensed MIT.

I know that's unorthodox, but the code is small enough that I feel comfortable including it in our larger GPL file.

You *can* include MIT code in a GPL project, but you typically put an explicit license header on the MIT *file* indicating the license and the copyright holder.

Here you link to the original code (good) and mention the author (good).  Mentioning the original license should be enough.